tracer/src/Datadog.Trace/SpanContext.cs

```diff
@@ -111,7 +111,7 @@ internal SpanContext(ulong? traceId, ulong spanId, int? samplingPriority, string
         /// <param name="rawTraceId">Raw trace id value</param>
         /// <param name="rawSpanId">Raw span id value</param>
         internal SpanContext(ISpanContext parent, TraceContext traceContext, string serviceName, ulong? traceId = null, ulong? spanId = null, string rawTraceId = null, string rawSpanId = null)
-            : this(parent?.TraceId ?? traceId, serviceName)
+            : this(parent?.TraceId > 0 ? parent.TraceId : traceId, serviceName)
```

Comment:
Note to reviewers: this change fixes a bug I found during testing. Parameter `traceId` should "Override the trace id if there's no parent" according to the xml comments above. This only worked is `parent == null`, but not when `parent.TraceId == 0`. This bug is triggered when using `new SpanContext(SpanContext.None, ...)`.